`inout unit[0] payload` array size must be larger than zero

I've tried a few different versions of slang `2025.6.1` and `2025.6.4` but am getting this error. Also seems like they have an issue in their repo to support 0 length arrays that's not implemented. How did you guys build this?

Build command used (from `install.bash` of `ever-training`): `CXX=$CXX CC=$CC cmake -DOptiX_INSTALL_DIR=$OptiX_INSTALL_DIR -DCMAKE_CUDA_ARCHITECTURES="50;60;61;70;75;80;86" ..
&& make -j8`

https://github.com/shader-slang/slang/issues/6816
and a test that says it's not possible
https://github.com/shader-slang/slang/blob/043278a527ab5744674417a08d924c67a60a486b/tests/diagnostics/array-zero-size.slang#L10

----
Output:

```
...
/ever_training/ever/splinetracers/fast_ellipsoid_splinetracer/../slang/optix-intrinsics.slang(618): error 30025: array size must be larger than zero.
      inout uint[0] payload,
                 ^
/ever_training/ever/splinetracers/fast_ellipsoid_splinetracer/../slang/optix.slang(15): error 39999: import of module 'optix_intrinsics' failed because of a compilation error
import optix_intrinsics;
       ^~~~~~~~~~~~~~~~
/ever_training/ever/splinetracers/fast_ellipsoid_splinetracer/slang/fast_shaders.slang(20): error 39999: import of module 'optix' failed because of a compilation error
import optix;
       ^~~~~
(0): fatal error 39999: compilation ceased
/ever_training/ever/splinetracers/fast_ellipsoid_splinetracer/slang/fast_shaders.slang(20): error 39999: import of module 'optix' failed because of a compilation error
import optix;
       ^~~~~
(0): fatal error 39999: compilation ceased
/ever_training/ever/splinetracers/fast_ellipsoid_splinetracer/../slang/optix-intrinsics.slang(618): error 30025: array size must be larger than zero.
      inout uint[0] payload,
                 ^
/ever_training/ever/splinetracers/fast_ellipsoid_splinetracer/../slang/optix.slang(15): error 39999: import of module 'optix_intrinsics' failed because of a compilation error
import optix_intrinsics;
       ^~~~~~~~~~~~~~~~
/ever_training/ever/splinetracers/fast_ellipsoid_splinetracer/slang/shaders.slang(20): error 39999: import of module 'optix' failed because of a compilation error
import optix;
       ^~~~~
(0): fatal error 39999: compilation ceased
/ever_training/ever/splinetracers/fast_ellipsoid_splinetracer/slang/shaders.slang(20): error 39999: import of module 'optix' failed because of a compilation error
import optix;
       ^~~~~
(0): fatal error 39999: compilation ceased
make[2]: *** [splinetracers/fast_ellipsoid_splinetracer/CMakeFiles/fast_ellipsoid_splinetracer_cpp_extension.dir/build.make:94: splinetracers/fast_ellipsoid_splinetracer/fast_ptx_code_file.ptx] Error 255
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [splinetracers/fast_ellipsoid_splinetracer/CMakeFiles/fast_ellipsoid_splinetracer_cpp_extension.dir/build.make:88: splinetracers/fast_ellipsoid_splinetracer/ptx_code_file.ptx] Error 255
make[1]: *** [CMakeFiles/Makefile2:128: splinetracers/fast_ellipsoid_splinetracer/CMakeFiles/fast_ellipsoid_splinetracer_cpp_extension.dir/all] Error 2
make: *** [Makefile:91: all] Error 2

ERROR conda.cli.main_run:execute(125): `conda run bash install.bash` failed. (See above for error)
LOOOK===========================
cmake version 3.31.6

CMake suite maintained and supported by Kitware (kitware.com/cmake).
-- The C compiler identification is GNU 11.4.0
-- The CXX compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using Python 3.10.16
-- Found torch 2.6.0+cu126
-- Using torch libraries: /opt/conda/envs/ever/lib/python3.10/site-packages/torch/lib
-- Using torch includes: /opt/conda/envs/ever/lib/python3.10/site-packages/torch/include;/opt/conda/envs/ever/lib/python3.10/site-packages/torch/include/torch/csrc/api/include;/opt/conda/envs/ever/lib/python3.10/site-packages/torch/include/TH;/opt/conda/envs/ever/lib/python3.10/site-packages/torch/include/THC
-- Using CUDA toolkit: /usr/local/cuda
-- Using CUDA architectures: 50;60;61;70;75;80;86
-- Using header-only CGAL
-- Targetting Unix Makefiles
-- Using /usr/bin/c++ compiler.
-- Found GMP: /usr/lib/x86_64-linux-gnu/libgmp.so
-- Found MPFR: /usr/lib/x86_64-linux-gnu/libmpfr.so
-- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.74.0/BoostConfig.cmake (found suitable version "1.74.0", minimum required is "1.48")
-- Boost include dirs: /usr/include
-- Boost libraries:    
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Using gcc version 4 or later. Adding -frounding-math
-- Clean Torch version: 2.6.0
-- Found Torch version: 2.6.0+cu126. CXX11ABI=1
-- pybind11 v2.11.1 
-- Found PythonInterp: /opt/conda/envs/ever/bin/python (found suitable version "3.10.16", minimum required is "3.6")
-- Found PythonLibs: /opt/conda/envs/ever/lib/libpython3.10.so
-- Performing Test HAS_FLTO
-- Performing Test HAS_FLTO - Success
-- Found CUDA: /usr/local/cuda (found suitable version "12.2", minimum required is "5.0")
-- Unsetting values associated with OptiX code generation
-- The CUDA compiler identification is NVIDIA 12.2.140 with host compiler GNU 11.4.0
-- Detecting CUDA compiler ABI info
-- Detecting CUDA compiler ABI info - done
-- Check for working CUDA compiler: /usr/local/cuda/bin/nvcc - skipped
-- Detecting CUDA compile features
-- Detecting CUDA compile features - done
-- Configuring done (31.2s)
-- Generating done (0.0s)
-- Build files have been written to: /ever_training/ever/build
[  6%] compile ptx from /ever_training/ever/splinetracers/fast_ellipsoid_splinetracer/shaders.cu
[ 12%] compile ptx from /ever_training/ever/splinetracers/fast_ellipsoid_splinetracer/slang/fast_shaders.slang
[ 18%] compile ptx from /ever_training/ever/splinetracers/fast_ellipsoid_splinetracer/slang/shaders.slang
[ 25%] embed ptx from /ever_training/ever/splinetracers/fast_ellipsoid_splinetracer/shaders.cu
```

--- a/splinetracers/slang/optix-intrinsics.slang
+++ b/splinetracers/slang/optix-intrinsics.slang
@@ -615,7 +615,6 @@ void optixTraceP0(
       in float3 ray_direction, //$2
       in float tmin, //$3
       in float tmax, //$4
-      inout uint[0] payload,
 )
 {
   __intrinsic_asm R"(

